projects/adxrs290-pmdz/src/adxrs290_iio.c

```diff
@@ -0,0 +1,150 @@
+
+#include "iio_types.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "adxrs290.h"
+#include "util.h"
+#include "error.h"
+
+#define NUM_CHANNELS 3
+
+/*
+ * Available cut-off frequencies of the low pass filter in Hz.
+ * The integer part and fractional part are represented separately.
+ */
+static const int adxrs290_lpf_3db_freq_hz_table[][2] = {
+	[0] = {480, 0},
+	[1] = {320, 0},
+	[2] = {160, 0},
+	[3] = {80, 0},
+	[4] = {56, 600000},
+	[5] = {40, 0},
+	[6] = {28, 300000},
+	[7] = {20, 0},
+};
+
+/*
+ * Available cut-off frequencies of the high pass filter in Hz.
+ * The integer part and fractional part are represented separately.
+ */
+static const int adxrs290_hpf_3db_freq_hz_table[][2] = {
+	[0] = {0, 0},
+	[1] = {0, 11000},
+	[2] = {0, 22000},
+	[3] = {0, 44000},
+	[4] = {0, 87000},
+	[5] = {0, 175000},
+	[6] = {0, 350000},
+	[7] = {0, 700000},
+	[8] = {1, 400000},
+	[9] = {2, 800000},
+	[10] = {11, 300000},
+};
+
+ssize_t get_adxrs290_iio_reg(void *device, char *buf, size_t len,
```

Comment:
Do direct_reg_access works?
When I implemented this, for me it worked like here: 
https://github.com/analogdevicesinc/no-OS/blob/aeb26133631dd51d55dd018b19697414875cc505/iio/iio_demo/demo_dev.c#L70

Reply:
Thanks for noticing this. You are right. My function is not working on direct_reg_access. It worked for iio_reg on writes but upon checking, it doesn't work if I try reading the same register twice. So I followed your approach and added the max register address.
I've tested the fix now in bot iio_reg and iio_attr via direct_reg_access.